### PR TITLE
Refactor repo to current monorepo standards

### PR DESCRIPTION
**Pull request by mattcompiles**
Changes
- Force prettier checks in CI
- Move CI from Travis to Github Actions
- Implement `preconstruct` build system
- Add `manypkg` linter
- Move release system from `lerna` to `changesets`
- Refactor webpack plugin to be ESM (TODO TypeScriptify...)
- Fix flaky gatsby plugin tests
- Probably some stuff I forgot to mention here...

### CODE REVIEW

File: packages/gatsby-plugin-treat/package.json [hunk 1]
```diff
@@ -16,5 +16,10 @@
   "dependencies": {
     "babel-plugin-treat": "^1.6.2"
   },
-  "license": "MIT"
+  "license": "MIT",
+  "devDependencies": {
+    "gatsby": "^2.18.11",
+    "treat": "*",
+    "wait-for-localhost": "^3.3.0"
```
Comment (michaeltaranto) [inline packages/gatsby-plugin-treat/package.json:23]: Looks like you removed the use of this library in the test, still needed elsewhere?

File: packages/gatsby-plugin-treat/tests/styling.test.js [hunk 2]
```diff
@@ -1,39 +1,70 @@
...
 async function startDevServer() {
-  gatsbyProcess = execa(`${gatsbyBinaryPath} develop -p ${gatsbyServerPort}`, gatsbyExecArgs);
-  gatsbyProcess.stdout.pipe(process.stdout);
-  gatsbyProcess.stderr.pipe(process.stdout);
+  gatsbyProcess = execa(
+    `${gatsbyBinaryPath} develop -p ${gatsbyServerPort}`,
+    gatsbyExecArgs,
+  );
+  // gatsbyProcess.stdout.pipe(process.stdout);
+  // gatsbyProcess.stderr.pipe(process.stdout);
```
Comment (michaeltaranto) [inline packages/gatsby-plugin-treat/tests/styling.test.js:56]: 🗑️

**Comment by maraisr** [comment]
WOW, thank you for this one @mattcompiles I can refactor the rollup plugin to take advantage of this new Adaptor method. Nice one mate.

File: site/webpack.config.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const { join } = require('path');
 const HtmlRenderWebpackPlugin = require('html-render-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const CopyPlugin = require('copy-webpack-plugin');
-const TreatPlugin = require('treat/webpack-plugin');
+const TreatPlugin = require('treat/webpack-plugin').default;
```
Comment (markdalgleish) [inline site/webpack.config.js:5]:
```suggestion
const { TreatPlugin } = require('treat/webpack-plugin');
```

**Review by markdalgleish — APPROVED** [review]

**Pull request merged**